filebeat/harvester/log.go

```diff
@@ -223,15 +220,33 @@ func (h *Harvester) updateOffset(increment int64) {
 	h.offset += increment
 }
 
-func (h *Harvester) UpdateState() {
+// SendStateUpdate send an empty event with the current state to update the registry
+func (h *Harvester) SendStateUpdate() {
 	logp.Debug("Update state: %s, offset: %v", h.Path, h.offset)
 	h.sendEvent(h.createEvent())
 }
 
+func (h *Harvester) GetState() input.FileState {
+	h.stateMutex.Lock()
+	defer h.stateMutex.Unlock()
+
+	return *h.State
+}
+
+// refreshState refreshes the values in State with the values from the harvester itself
+func (h *Harvester) refreshState() {
+	h.stateMutex.Lock()
+	defer h.stateMutex.Unlock()
+
+	h.State.Source = h.Path
+	h.State.Offset = h.getOffset()
+	h.State.FileStateOS = input.GetOSFileState(h.State.Fileinfo)
+}
+
 func (h *Harvester) close() {
 	// On completion, push offset so we can continue where we left off if we relaunch on the same file
-	if h.Stat != nil {
-		h.Stat.Offset <- h.getOffset()
+	if h.State != nil {
+		h.State.OffsetChannel <- h.getOffset()
```

Comment:
how about `LastKnownOffset` instead of `OffsetChannel`. We already tripped a few times being confused why channel must be full in order to fileState to be finished.